src/Datadog.Trace/Configuration/StringConfigurationSource.cs

```diff
@@ -76,11 +77,8 @@ public abstract class StringConfigurationSource : IConfigurationSource
         /// <inheritdoc />
         public virtual bool? GetBool(string key)
         {
-            string value = GetString(key);
-
-            return bool.TryParse(value, out bool result)
-                       ? result
-                       : (bool?)null;
+            var value = GetString(key);
+            return value?.ToBoolean();
```

Comment:
`ToBoolean` can throw an exception, whereas `TryParse` cannot? Is there a reason you're switching the implementation?

Reply:
It will only throw if it's null, right?
TryParse did not cover all cases like `1` and `0`
This test data fails without this change: https://github.com/DataDog/dd-trace-dotnet/pull/654/files#diff-7dc0af97db71be8e20b85bd741ed31a0R17

Reply:
Ah I misread the code. I mistook this `ToBoolean` with the framework's `Convert.ToBoolean` call instead of our own extension method. This looks good 👍